File: array/2Q1错误的集合.py
from typing import List
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        nums=sorted(nums)
        n=len(nums)
        for i in range(0,n):
            if nums[i]!=i+1:   
                return [nums[i],i+1]
        return []

File: array/2Q1错误的集合.cpp
#include <unordered_map>
#include <set>
#include <vector>
using namespace std;
class Solution {
    public:
        vector<int> findErrorNums(vector<int>& nums) {
            set<int>hav;
            int n=nums.size();
            vector<int>ans(2,0);
            for(int i=0;i<n;i++){
                if(hav.find(nums[i])!=hav.end()){
                    ans[0]=nums[i];
                }
                hav.emplace(nums[i]);
            }
            for(int i=1;i<=n;i++){
                if(hav.find(i)==hav.end()){
                    ans[1]=i;
                    return ans;
                }
            }
            return {};
        }
    };